### PR TITLE
update(content): add a link to the github repo

### LINKED ISSUE #804: There's no link to our GitHub repo from the Documentation site
**Describe the bug**

We don't have a link to https://github.com/gomods/athens (this repo 😄) from the [docs site](https://docs.gomods/io)

**Expected behavior**

We should have a link at least on the home page, so that people who want to get started, look at code, contribute, etc... can do so

### PR DESCRIPTION
**Pull request by hp-plus5**
**What is the problem I am trying to address?**

There was an issue request for a link to the Github repo to be added to the docs.

**How is the fix applied?**

I added a line of content.

**Mention the issue number it fixes or add the details of the changes if it doesn't has a specific issue.**

Fixes #804

### CODE REVIEW

File: docs/content/intro/_index.md [hunk 1]
```diff
@@ -15,3 +15,4 @@ We recommend you read this section from front to back:
 - [Athens 101](./first-content) - the basic mechanics of Athens
 - [Download Protocol](./protocol) - the fundamental way that Athens serves Go packages
 - [Components](./components) - the different moving pieces of Athens
+- [Source Code](https://github.com/gomods/athens) - the source code of Athens
```
[the remark this answers is not in the document]
Comment (arschles) [reply docs/content/intro/_index.md]: Is there any way we can make this more prominent? Like [this](https://blog.github.com/2008-12-19-github-ribbons/)?

**Comment by michalpristas** [comment]
Hey, @hp-plus5 please let us know if you ran into some troubles so we can help you to get this in.

File: docs/content/intro/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (arschles) [reply docs/content/intro/_index.md]: @hp-plus5 let me know what you think about this?

[the remark this answers is not in the document]
Comment (hp-plus5) [reply docs/content/intro/_index.md]: Sorry for the absence. Do you think we should keep in the 'source code' where it is AND add a banner, or just have a banner? Should we put it on every page or just the intro?

[the remark this answers is not in the document]
Comment (arschles) [reply docs/content/intro/_index.md]: @hp-plus5 no problem for my absence too! I think for now, the banner and the link are both fine to just go on the front page

**Pull request closed without merging**